gatsby-plugin-webfonts/src/modules/selfHosted.js

```diff
@@ -0,0 +1,69 @@
+import postcss from "postcss";
+import postcssJs from "postcss-js";
+import path from "path";
+import { constants, copyFileSync, existsSync, mkdirSync } from "fs";
+
+const defaultFontOptions = {
+  fontDisplay: `swap`,
+};
+
+export default function selfHosted({ directory, pathPrefix = `` }, reporter) {
+  const getFontFace = async (font) => {
+    const { family, urls, ...cssProperties } = createFontOptions(font);
+
+    const src = Object.entries(urls).map(([format, url]) => {
+      const sourcePath = path.join(directory, url);
+      if (!existsSync(sourcePath)) {
+        reporter.panicOnBuild(
+          `Specified selfHosted font file missing: "${sourcePath}"`,
+        );
+        return ``;
+      }
+
+      const fileName = path.basename(url);
+      const cssDir = path.join(`/static`, `webfonts`);
```

Comment:
The path prefix must be considered. You can test it with path-prefix-prod-runtime (yarn build && yarn serve) => GET http://localhost:9000/static/webfonts/OpenSans300.woff2 404 (Not Found)

Reply:
I have now got it working and fully tested with path prefixes.